stripe/_http_client.py

```diff
@@ -462,17 +463,45 @@ async def sleep_async(cls: Type[Self], secs: float) -> Awaitable[None]:
             "HTTPClientAsync subclasses must implement `sleep`"
         )
 
+    @overload
     async def _request_with_retries_internal_async(
         self,
-        method,
-        url,
-        headers,
+        method: str,
+        url: str,
+        headers: Mapping[str, str],
         post_data,
-        is_streaming,
-        max_network_retries,
+        is_streaming: Literal[False],
```

Comment:
Why do we need this overload? Is this not the same as `is_streaming: bool` in the original method definition?

Reply:
pyright complains if you only have a single overload. The other overload has a more narrow return type so if you want that one you have to include this one as well.